src/System.Text.Json/src/System/Text/Json/Node/JsonNode.Traversal.cs

```diff
@@ -0,0 +1,327 @@
+
+                    if (currentFrame.ValueKind == JsonValueKind.Object)
+                    {
+                        writer.WriteEndObject();
+                    }
+                    if (currentFrame.ValueKind == JsonValueKind.Array)
+                    {
+                        writer.WriteEndArray();
+                    }
+
+                    continue;
+                }
+
+                if (currentFrame.PropertyName != null)
+                {
+                    writer.WritePropertyName(currentFrame.PropertyName);
+                }
+
+                switch (currentFrame.PropertyValue)
+                {
+                    case JsonObject jsonObject:
+                        writer.WriteStartObject();
+
+                        // Add end of object marker:
+                        recursionStack.Push(new RecursionStackFrame(null, null, JsonValueKind.Object));
+
+                        // Add properties to recursion stack. Reverse enumerator to keep properties order:
+                        foreach (KeyValuePair<string, JsonNode> jsonProperty in jsonObject.Reverse())
+                        {
+                            recursionStack.Push(new RecursionStackFrame(jsonProperty.Key, jsonProperty.Value));
+                        }
+                        break;
+                    case JsonArray jsonArray:
+                        writer.WriteStartArray();
+
+                        // Add end of array marker:
+                        recursionStack.Push(new RecursionStackFrame(null, null, JsonValueKind.Array));
+
+                        // Add items to recursion stack. Reverse enumerator to keep items order:
+                        foreach (JsonNode item in jsonArray.Reverse())
+                        {
+                            recursionStack.Push(new RecursionStackFrame(null, item));
+                        }
+                        break;
+                    case JsonString jsonString:
+                        writer.WriteStringValue(jsonString.Value);
+                        break;
+                    case JsonNumber jsonNumber:
+                        writer.WriteNumberValue(Encoding.UTF8.GetBytes(jsonNumber.ToString()));
+                        break;
+                    case JsonBoolean jsonBoolean:
+                        writer.WriteBooleanValue(jsonBoolean.Value);
+                        break;
+                    case JsonNull _:
+                        writer.WriteNullValue();
+                        break;
+                }
+
+                writer.Flush();
+            }
+
+            writer.Flush();
+        }
+
+        /// <summary>
+        ///   Converts the current instance to string in JSON format.
+        /// </summary>
+        /// <returns>JSON representation of current instance.</returns>
+        public string ToJsonString()
+        {
+            var stream = new MemoryStream();
```

Comment:
I believe we should also wrap this MemoryStream inside a using statement since it is also disposable.

Reply:
I changed it to ArrayBufferWriter anyway.